specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/compute.json

```diff
@@ -3439,6 +3439,48 @@
         "x-ms-long-running-operation": true
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/SimulateEviction": {
+      "post": {
+        "tags": [
+          "VirtualMachines"
+        ],
+        "operationId": "VirtualMachines_SimulateEviction",
+        "description": "The operation to simulate the eviction of spot virtual machine",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "vmName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the virtual machine."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
```

Comment:
Why use a  `no content` response?  How does the user know the result of the request?

Reply:
@KrisBash, this is the API to Simulate the Eviction in same way as how Spot Virtual Machine is evicted due to capacity or price reasons. The success from this API means, the request is sent successfully to the backend, and once eviction is triggered, customer can get a 30 seconds in-vm notification via Azure metadata service. Basically, this API's success response basically mean that that request is received successfully and no other payload is needed in the response.

And, HTTP Status 204 (No Content) indicates that the server has successfully fulfilled the request and that there is no content to send in the response payload body. Therefore it makes more sense return HTTP Status 200 (OK)